### ISSUE TITLE
The compressed file cannot be decompressed by lzma command on Mac OS

### ISSUE THREAD

**Issue by ddaddca** [nmrugg/LZMA-JS#44]
I'm using LZMA-JS on nodejs, my code is as below

```
var my_lzma = require("lzma");
var fs = require('fs');

var result = my_lzma.compress('123456789qwertyuiop');
fs.writeFileSync('b.txt', new Buffer(result, 'ascii'), null, 'b');
```

then, I renamed "b.txt" to "b.txt.lzma" and tried to use lzma command to decompress it.  but I only get the error message

> lzma: b2.txt.lzma: Compressed data is corrupt

LZMA-JS and lzma command both are based on the same algorithm, from my understanding, lzma can decompress the file that is compressed by LZMA-JS, am I right?  So could you please help me to find out   the problem that why the file get decompressed failed? Thanks

**Comment by addaleax** [comment]
I can reproduce this on Linux with all versions published to npm, including 1.0.0, so yeah, I think this is a bug here

**Comment by addaleax** [comment]
I’d prefer not to do anything without @nmrugg’s input but I think LZMA-JS is wrongly writing the end-of-payload marker to the compressed stream even though the compressed size is known. That should not be allowed according to the docs ([ref](https://github.com/addaleax/xz/blob/faf302137e54d605b44ecf0373cb51a6403a2de1/doc/lzma-file-format.txt#L127)), and commenting out the call to `$WriteEndMarker` seems to solve this problem.

**Comment by nmrugg** [comment]
Good call on `$WriteEndMarker`. I had originally tested against LZMA SDK 4.43, which doesn't mind unnecessary `$WriteEndMarker`.

Right now, we don't actually stream, so we always know the length, so we can't use `$WriteEndMarker`, right?

**Comment by addaleax** [comment]
Yep, until there’s stream support, it’s probably best to comment it out. And when that’s coming, it needs to be enabled depending on some kind of flag that indicates whether the length has been written or not.

**Comment by bilthon** [comment]
So has this issue been solved? I'm running in the exact same problem here, can't decompress something in the mac command line.

**Comment by jeanbmar** [comment]
I'm late to the party but you can do this:
```javascript
const lzmajs = require('lzma');
lzmajs.LZMA().disableEndMark = true;
// ... do your stuff
```

**Comment by markusmobius** [comment]
Thank you @jeanbmar !

I spent several hours on being unable to decompress LZMA data from LZMA-JS on Python:

`json=lzma.decompress(data=data, format=lzma.FORMAT_ALONE)
`
But as soon as I set the `disableEndMark ` to true it worked!
